app/services/move_flag.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class MoveFlag
+  include Service::Base
+
+  contract
+  model :flag
+  policy :invalid_access
+  policy :invalid_move
+
+  transaction do
+    step :move
+    step :log
+  end
+
+  class Contract
+    attribute :flag_id, :integer
+    attribute :direction, :string
+    validates :flag_id, presence: true
+    validates :direction, presence: true
+  end
+
+  private
+
+  def fetch_flag(flag_id:)
+    Flag.find(flag_id)
+  end
+
+  def invalid_access(guardian:, flag:)
+    guardian.can_move_flag?(flag)
+  end
+
+  def all_flags
+    @all_flags ||= Flag.where(score_type: false).where.not(name_key: "notify_user").order(:position)
```

Comment:
I wonder if `.where(score_type: false)` should be a default scope for flags? Or is this in particular a special case where we don't want them?